syntax = "proto3";
package steward;

import "proto/steward/user.proto";
import "proto/steward/maintenance.proto";
import "proto/steward/asset.proto";
import "proto/steward/schedule.proto";

service UserService {
  rpc GetUser(GetUserRequest) returns (User) {} ;
  rpc CreateUser(CreateUserRequest) returns (User) {} ;
  rpc DeleteUser(DeleteUserRequest) returns (User) {} ;
  rpc UpdateUser(UpdateUserRequest) returns (User) {} ;
  rpc ListUsers(ListUsersRequest) returns (stream User) {} ;
}

service MaintenanceService {
  rpc GetMaintenance(GetMaintenanceRequest) returns (Maintenance) {} ;
  rpc CreateMaintenance(Maintenance) returns (Maintenance) {} ;
  rpc DeleteMaintenance(DeleteMaintenanceRequest) returns (Maintenance) {} ;
  rpc UpdateMaintenance(UpdateMaintenanceRequest) returns (Maintenance) {} ;
  rpc ListMaintenances(ListMaintenancesRequest) returns (stream Maintenance) {} ;
}

service AssetService {
  rpc GetAsset(GetAssetRequest) returns (Asset) {} ;
  rpc CreateAsset(Asset) returns (Asset) {} ;
  rpc DeleteAsset(DeleteAssetRequest) returns (Asset) {} ;
  rpc UpdateAsset(UpdateAssetRequest) returns (Asset) {} ;
  rpc ListAssets(ListAssetsRequest) returns (stream Asset) {} ;
}

service ScheduleService {
  rpc GetSchedule(GetScheduleRequest) returns (Schedule) {} ;
  rpc CreateSchedule(Schedule) returns (Schedule) {} ;
  rpc DeleteSchedule(DeleteScheduleRequest) returns (Schedule) {} ;
  rpc UpdateSchedule(UpdateScheduleRequest) returns (Schedule) {} ;
  rpc ListSchedules(ListSchedulesRequest) returns (stream Schedule) {} ;
}
